### PR TITLE
Update Badges and the count Badge component for dark mode support

### PR DESCRIPTION
**Pull request by KATIETOLER**
## Description & motivation 💭 

Badge audit for darkmode:
from [DT-1749](https://temporalio.atlassian.net/browse/DT-1749)

Badge count audit for darkmode:
from [DT-1750](https://temporalio.atlassian.net/browse/DT-1750)

### Design Considerations 🎨 
Made to match [this design](https://www.figma.com/file/LNN3Cu8Akqb240WmJmbtsS/Holocene?type=design&node-id=3128-136742&mode=design&t=6kh47HuCPzHhdqHt-0), screenshot below 👇

image of designs

### Screenshots  📸 
Screenshot 2024-03-19 at 2 56 44 PM
Screenshot 2024-03-19 at 2 56 49 PM
Screenshot 2024-03-19 at 2 57 07 PM
Screenshot 2024-03-19 at 2 55 39 PM

### How was this tested 👻 
Storybook

### CODE REVIEW

File: src/lib/holocene/__snapshots__/button.test.ts.snap
Comment (rossedfort) [inline src/lib/holocene/__snapshots__/button.test.ts.snap:1]: You may need to merge main into your branch, I deleted the button snapshot tests

File: src/lib/holocene/badge.svelte [hunk 1]
```diff
@@ -41,26 +41,26 @@
...
   .gray {
-    @apply bg-badge text-primary;
+    @apply bg-badge text-primary dark:bg-slate-100 dark:text-black;
   }
 
   .active,
   .available,
   .green {
-    @apply border-green-700 bg-green-100 text-green-700;
+    @apply border-green-700 bg-[#71D375] text-black;
```
Comment (rossedfort) [inline src/lib/holocene/badge.svelte:60]: Does `#71D375` replace our existing green-100? I'd rather not use one off colors and have this added to the theme instead.

**Comment by tigernaut** [comment]
lgtm, we should consider updating the badge names in the future

File: src/lib/holocene/badge.svelte (hunk 1 above)
Comment (KATIETOLER) [reply src/lib/holocene/badge.svelte:60]: Oh yes, I definitely meant to delete this before I pushed, design is still working on this one. Thank you.

**Review by GraceGardner — APPROVED** [review]
Looks good, thanks for resolving the issues Ross mentioned!

**Comment by rossedfort** [comment]
@GraceGardner I probably should have "requested changes" but the reason I hadn't approved this yet is cause the button snapshots was still showing as an added file. @KATIETOLER can you create a new PR to remove it please?

**Pull request merged**